openpype/hosts/maya/plugins/load/load_reference.py

```diff
@@ -205,6 +209,12 @@ def process_reference(self, context, name, namespace, options):
                 self._post_process_rig(name, namespace, context, options)
             else:
                 if "translate" in options:
+                    if not attach_to_root and new_nodes:
+                        reference_node = get_reference_node(new_nodes)
+                        for node in new_nodes:
+                            if node != reference_node:
+                                break
+                        group_name = node.lstrip("|")
```

Comment:
This is not the way. Without the leading `|` it'll allow to match ANY node name with that name, and there can be clashes. Technically the behavior we're using is that it contains a namespaces at the very least, and we uniqify the namespace somewhere earlier in the code so it's basically a zero percent chance, but still it's odd logic doing it this way.

I feel like we're better off just getting the top level nodes from the returned nodes. I did a quick test in Maya 2024 here but it includes the group name as well if the group is generated with `groupReference=True` and `groupName="whatever"`.

Since we know that basically directly after the referencing the loaded content is NOT parented to anything else (the reference or its group is top-level maya node) we know we can just get the group name via:

```python
root_nodes = cmds.ls(new_nodes, assemblies=True, long=True)
group_name = root_nodes[0]  # we assume only a single root is ever loaded
```

As such, from the beginning `group_name` can always be retrieved like that and doesn't need altering behavior between the if else branches.
 
Like with the current logic we assume only a single root is ever loaded, be aware that this isn't necessarily true if NO grouping was enabled for the reference and the loaded content, e.g. an Alembic reference actually has multiple top level nodes. This also means that if the case happens where translating like this will happen that only ONE of the root nodes will ever be translated (as opposed to all of them.)

Reply:
Ok, makes sense.